packetbeat/protos/nfs/nfs4.go

```diff
@@ -165,20 +165,37 @@ func (nfs *Nfs) eatData(op int, xdr *Xdr) {
 	}
 }
 
-func (nfs *Nfs) getV4Opcode(xdr *Xdr) string {
+// Find main operation in a compound call.
+//
+//  Compound request group multiple nfs operations into a single request. Nevertheless, all compound requests are
+// triggered by  end-user activity, line 'ls', 'open', 'stat' and IO calls. Depending on which operations are combined
+//  tougher the main operation can be different. For example, in compound:
+//
+// PUTFH + READDIR + GETATTR
+//
+// READDIR is the main operation. while in
+//
+// PUTFH + GETATTR
+//
+// GETATTR is the main operation.
+//
+//  This  function tries to guess main purpose of the request. If not found, then the last operation is used.
```

Comment:
Technically, one can build a compound with READDIR and WRITE in it. In practice, nfs clients don't do that, as all nfs requests triggered by  user activities, and there are no posix call that will trigger readdir+write. Well, synthetic clients can....but they used only for server functional testing.